xline/src/server/maintenance.rs

```diff
@@ -67,9 +81,32 @@ where
         &self,
         _request: tonic::Request<StatusRequest>,
     ) -> Result<tonic::Response<StatusResponse>, tonic::Status> {
-        Err(tonic::Status::unimplemented(
-            "status is unimplemented".to_owned(),
-        ))
+        let cluster = self
+            .client
+            .get_cluster_from_curp(false)
+            .await
+            .map_err(propose_err_to_status)?;
+        let header = self.header_gen.gen_header();
+        let self_id = self.cluster_info.self_id();
+        let is_learner = cluster
+            .members
+            .iter()
+            .find(|member| member.id == self_id)
+            .map_or(false, |member| member.is_learner);
+        // TODO should we add `raft_index` and `raft_applied_index` in FetchClusterResponse?
+        let response = StatusResponse {
+            header: Some(header),
+            version: "NA".to_owned(),
```

Comment:
`version` should be Xline version. Please refer to etcd and fill other fields of this repsonse.

Reply:
done